betterproto/tests/test_get_ref_type.py

```diff
@@ -77,6 +79,227 @@ def test_importing_google_wrappers_unwraps_them(google_type: str, expected_name:
 def test_importing_google_wrappers_without_unwrapping(
     google_type: str, expected_name: str
 ):
-    name = get_ref_type(package="", imports=set(), type_name=google_type, unwrap=False)
+    name = get_ref_type(
+        package="", imports=set(), source_type=google_type, unwrap=False
+    )
 
     assert name == expected_name
+
+
+def test_import_child_package_from_package():
+    imports = set()
+    name = get_ref_type(
+        package="package", imports=imports, source_type="package.child.Message"
+    )
+
+    assert imports == {"from . import child"}
+    assert name == "child.Message"
+
+
+def test_import_child_package_from_root():
+    imports = set()
+    name = get_ref_type(package="", imports=imports, source_type="child.Message")
+
+    assert imports == {"from . import child"}
+    assert name == "child.Message"
+
+
+def test_import_camel_cased():
+    imports = set()
+    name = get_ref_type(
+        package="", imports=imports, source_type="child_package.example_message"
+    )
+
+    assert imports == {"from . import child_package"}
+    assert name == "child_package.ExampleMessage"
+
+
+def test_import_nested_child_from_root():
+    imports = set()
+    name = get_ref_type(package="", imports=imports, source_type="nested.child.Message")
+
+    assert imports == {"from .nested import child as nested_child"}
+    assert name == "nested_child.Message"
+
+
+def test_import_deeply_nested_child_from_root():
```

Comment:
I find "..._from_root" a little confusing, like root implies sys.path...

I would think of this as importing from sibling, as this present module and the module that it's importing from share the same parent package.

I would think of what is described below as import_from_sibling instead as not importing at all, unless I've misunderstood?

Reply:
thanks for your comments.

**test_import_deeply_nested_child_from_root**

This test is about importing a deeply nested package from the context of the top-level package (no package).

```python
def test_import_deeply_nested_child_from_root():
    imports = set()
    name = get_ref_type(
        package="", imports=imports, source_type="deeply.nested.child.Message"
    )

    assert imports == {"from .deeply.nested import child as deeply_nested_child"}
    assert name == "deeply_nested_child.Message"
```

as you can see from the generated import, it imports from 2 levels lower, so they are not siblings.

**import_from_sibling**

when a type is referenced from a 'sibling' (same package), then indeed, an import statement is not necessary.

maybe i should rename `import_*` to `reference_*`, because its about referencing types, that sometimes lead to imports.

agree?

Reply:
by rename, i also mean the functions in the library code

Reply:
Changes done as a result of these comments:
- replaced method comments with functionally descriptive docstrings
- terminology: `reference` instead of `import`
- added asserts for empty import set cases

I'm not sure what to do about `_from_root`. I understand it can be confused with sys.path.
The context from which the import is done, is the root of the module, of which all packages for the generated code are descendants. It's the `root` from the perspective of the protobuf package hierarchy.

Would be great if we can find a term for that, that is not confusing.